src/modules/compliance/src/lib/Procedure.cpp

```diff
@@ -5,52 +5,147 @@
 
 #include "Logging.h"
 
+#include <cassert>
 #include <parson.h>
-#include <sstream>
 
 namespace compliance
 {
+namespace
+{
+std::size_t SkipSpaces(const std::string& input, std::size_t pos)
+{
+    while (pos < input.size() && isspace(input[pos]))
+    {
+        ++pos;
+    }
+    return pos;
+}
+
+Result<std::size_t> ParseKey(const std::string& input, std::size_t pos)
+{
+    bool first = true;
+    while (pos < input.size() && !isspace(input[pos]) && input[pos] != '=')
```

Comment:
>while (pos < input.size() && !isspace(input[pos]) && input[pos] != '=') [](http://example.com/codeflow?start=4&length=71)

Add braces to separate conditions? Like the previous comment, I will stop commenting on this in this PR.